src/assocWith.js

```diff
@@ -0,0 +1,11 @@
+const assoc    = require('ramda/src/assoc')
+const curry    = require('ramda/src/curry')
+const lensProp = require('ramda/src/lensProp')
+const over     = require('ramda/src/over')
+const identity = require('ramda/src/identity')
+
+// assocWith : String -> ({ k: v } -> a) -> { k: v }
+const assocWith = (key, fn) =>
+  over(lensProp(key), fn)
```

Comment:
This doesn't do what you want it to.  Ramda's [over](http://devdocs.io/ramda/index#over) passing the focused value of the lens to the transforming function, not the original value.